### ISSUE TITLE
First split the data, then normalize

### ISSUE THREAD

**Issue by fabianjkrueger** [fabianjkrueger/RestAI#2]
# first split
X_train, X_test = train_test_split(X, test_size=0.2, random_state=42)

# then fit scaler only on training data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)

# apply same transformation to test data
X_test_scaled = scaler.transform(X_test)

**Comment by fabianjkrueger** [comment]
# save the scaler for later use
from joblib import dump
dump(scaler, "scaler.pkl")

# when predicting new data
from joblib import load
scaler = load("scaler.pkl")
X_new_scaled = scaler.transform(X_new)

**Comment by fabianjkrueger** [comment]
Also, I need a dictionary vectorizer for inference to make the queried data accessible to the model
